Libs/GObject-2.0/Native/Classes/ObjectWrapper.cs

```diff
@@ -27,7 +33,14 @@ public static class ObjectWrapper
             // We need to find the "true type" of the object so that the user
             // can upcast/downcast as necessary. Retrieve the gtype from the
             // object's class struct.
-            Type gtype = GetTypeFromHandle(handle);
+
+            Type gtype = GetTypeFromInstance(handle);
+
+            Debug.Assert(
```

Comment:
I think we do not need this assert as the `GetTypeFromInstance` method retrievs the type from the handle. The old implementation was just a bug. Otherwise we could write debug asserts for all kind of in between steps.

Reply:
I like having Debug asserts as it helps us ensure that the type integration's state is exactly how we expect it to be. At the same time, unlike exceptions it won't degrade performance on a release build which is another benefit.

I'd be in favour of writing more debug asserts in the type integration part of GObject so we don't have any surprises.

Reply:
Okay, we can keep it as it is not hurting like you said.